packages/app/lib/app.dart

```diff
@@ -3,14 +3,47 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:shared_preferences/shared_preferences.dart';
 
 import 'package:app/router.dart';
 import 'package:app/ui/widgets/image_provider.dart';
 import 'package:app/io/graphql/graphql.dart' as graphql;
 
-class MeliApp extends StatelessWidget {
+class MeliApp extends StatefulWidget {
   MeliApp({super.key});
 
+  @override
+  State<MeliApp> createState() => MeliAppState();
+}
+
+class MeliAppState extends State<MeliApp> {
+  final Future<SharedPreferences> _prefs = SharedPreferences.getInstance();
+  Locale? _locale;
+
+  @override
+  void initState() {
+    super.initState();
+    _prefs.then((SharedPreferences prefs) {
+      final String localString = prefs.getString('locale') ?? 'pt';
+      setState(() {
+        _locale = Locale(localString);
+      });
+    });
+  }
+
+  Future<bool> changeLocale(Locale locale) async {
+    final SharedPreferences prefs = await _prefs;
+    bool _success = await prefs.setString('locale', locale.toString());
+
+    if (_success) {
+      setState(() {
+        _locale = locale;
+      });
+    }
+
+    return _success;
+  }
```

Comment:
Could make sense to set that state as an argument to create `MeliApp` (one can make it stateless again then) and retrieve it in our main process (while the splash screen is shown)

Reply:
I'll make a ticket!